src/main/java/redis/clients/util/JedisClusterHashTagUtil.java

```diff
@@ -0,0 +1,45 @@
+package redis.clients.util;
+
+/**
+ * Holds various methods/utilities to manipualte and parse redis hash-tags.
+ * See <a
+ * href="http://redis.io/topics/cluster-spec">Cluster-Spec : Keys hash tags</a>
+ */
+public final class JedisClusterHashTagUtil {
+
+  private JedisClusterHashTagUtil() {
+    throw new InstantiationError("Must not instantiate this class");
+  }
+
+  public static String getHashTag(String key) {
+    return extractHashTag(key, false, false);
+  }
+
+  public static boolean isClusterCompliantMatchPattern(String matchPattern) {
+    String tag = extractHashTag(matchPattern, true, true);
+    if (tag != null && !tag.isEmpty() && matchPattern.startsWith(tag)) {
+      return matchPattern.substring(tag.length()).equals("*");
```

Comment:
Sorry but that's not the case.
You can test manually and see that running the scan only ( without the '*' globbing ) yields no results . My string match tests behaves as expected and is the right way to scan with Tags within a clustered setup ( Though I think the same goes for standard SCANing )

http://redis.io/commands/scan -   'It is possible to only iterate elements matching a given glob-style pattern'